spec/models/miq_provision_request_spec.rb

```diff
@@ -201,6 +201,12 @@ def create_request(user, vm_template, prov_options)
               {:count => 12, :memory => 8_589_938_688, :cpu => 32, :storage => 44.gigabytes}
             end
             it_behaves_like "check_quota"
+
+            it "invalid vm_template does not raise error" do
+              create_requests
+              MiqRequest.first.update_attributes(:vm_template => nil)
+              expect { request.check_quota(quota_method) }.not_to raise_error
```

Comment:
This isn't correct since MiqProvisionRequest has a validation on source (vm_template) presence. It will definitely raise an error.